proxy/proxy.go

```diff
@@ -406,10 +407,29 @@ func cloneHeaderExcluding(h http.Header, excludeList map[string]bool) http.Heade
 	return hh
 }
 
+// The sync.Pool's New function should generally only return pointer types,
+// since a pointer can be put into the return interface value without an allocation.
+//
+// It is fine to store slice as per https://github.com/golang/go/issues/16323
+// but staticcheck complains about non-pointer type (https://staticcheck.io/docs/checks#SA6002)
+// therefore slice pointer is used.
+//
+// Note: sync.Pool example suggests to use bytes.Buffer that can dynamically grow and thus
+// cause problems, see https://github.com/golang/go/issues/23199 for details.
+var bufferPool = sync.Pool{
```

Comment:
Can you change it to non package global?
It’s a bad thing that we used it and we should omit it if we can.

Reply:
It is possible to make an instance per proxy. What is the problem with single shared pool (even for multiple proxies)?

Reply:
Another thing is that maybe it should be under the flag to turn it on/off quickly to experiment of if there is some issue with it in real setup...

Reply:
Everytime we had globals it turned out to be bad. We should avoid creating them.
Of course we have globals that we should some day get rid of.

Reply:
Will fix it, no problem, was just genuinely interested what could be the possible problem with it.

Reply:
One typical regret is when testing a previously unanticipated scenario.

Though true, that a bufferpool is kind of just a reference to the process's actual memory pool, and no actual problem comes to my mind right now, but still, I also prefer to just avoid globals, especially as it's easy enough.